packages/components/view-switcher/src/view-switcher.spec.js

```diff
@@ -131,6 +132,34 @@ describe('rendering', () => {
     screen.getByLabelText('Test Button 1').click();
     expect(handleClick).not.toHaveBeenCalled();
   });
+
+  it('should be controlled when selectedValue is passed', () => {
+    const handleClick = jest.fn();
+    function TestComponent() {
+      const [seletedValue, setSelectedValue] = useState('test-button-1');
+
+      return (
+        <Group selectedValue={seletedValue} onChange={setSelectedValue}>
+          <Button value="test-button-1" onClick={handleClick}>
+            Test Button 1
+          </Button>
+          <Button value="test-button-2" onClick={handleClick}>
+            Test Button 2
+          </Button>
+        </Group>
+      );
+    }
+    render(<TestComponent />);
+
+    screen.getByLabelText('Test Button 1').click();
+    expect(handleClick).not.toHaveBeenCalled();
```

Comment:
To make this test more clear, maybe we should explicitly pass the initial value so that it's clear that the click handler won't be called.

```suggestion
    render(<TestComponent selectedValue="test-button-1" />);

    screen.getByLabelText('Test Button 1').click();
    expect(handleClick).not.toHaveBeenCalled();
```

Reply:
You say "explicitly pass the initial" then pass the "selectedValue". Not sure I follow. We now will have a warning if we pass both. You're saying we should pass different defaultSelected and selectedValue to then assert that the default isn't used, right?

Reply:
I added a commit. Maybe you adjust it if you thought otherwise. Thanks.